libsolidity/ast/ASTJsonImporter.cpp

```diff
@@ -0,0 +1,954 @@
+		nullOrASTString(_node, "documentation"),
+		createParameterList(member(_node, "parameters")),
+		modifiers,
+		createParameterList(member(_node, "returnParameters")),
+		memberAsBool(_node, "implemented") ? createBlock(member(_node, "body")) : nullptr
+	);
+}
+
+ASTPointer<VariableDeclaration> ASTJsonImporter::createVariableDeclaration(Json::Value const& _node)
+{
+	return createASTNode<VariableDeclaration>(
+		_node,
+		nullOrCast<TypeName>(member(_node, "typeName")),
+		make_shared<ASTString>(member(_node, "name").asString()),
+		nullOrCast<Expression>(member(_node, "value")),
+		visibility(_node),
+		memberAsBool(_node, "stateVariable"),
+		_node.isMember("indexed") ? memberAsBool(_node, "indexed") : false,
+		memberAsBool(_node, "constant"),
+		_node["overrides"].isNull() ? nullptr : createOverrideSpecifier(member(_node, "overrides")),
+		location(_node)
+	);
+}
+
+ASTPointer<ModifierDefinition> ASTJsonImporter::createModifierDefinition(Json::Value const&  _node)
+{
+	return createASTNode<ModifierDefinition>(
+		_node,
+		memberAsASTString(_node, "name"),
+		nullOrASTString(_node,"documentation"),
+		createParameterList(member(_node, "parameters")),
+		memberAsBool(_node, "virtual"),
+		_node["overrides"].isNull() ? nullptr : createOverrideSpecifier(member(_node, "overrides")),
+		createBlock(member(_node, "body"))
+	);
+}
+
+ASTPointer<ModifierInvocation> ASTJsonImporter::createModifierInvocation(Json::Value const&  _node)
+{
+	std::vector<ASTPointer<Expression>> arguments;
+	for (auto& arg: member(_node, "arguments"))
+		arguments.push_back(convertJsonToASTNode<Expression>(arg));
+	return createASTNode<ModifierInvocation>(
+		_node,
+		createIdentifier(member(_node, "modifierName")),
+		member(_node, "arguments").isNull() ? nullptr : make_unique<std::vector<ASTPointer<Expression>>>(arguments)
+	);
+}
+
+ASTPointer<EventDefinition> ASTJsonImporter::createEventDefinition(Json::Value const&  _node)
+{
+	return createASTNode<EventDefinition>(
+		_node,
+		memberAsASTString(_node, "name"),
+		nullOrASTString(_node, "documentation"),
+		createParameterList(member(_node, "parameters")),
+		memberAsBool(_node, "anonymous")
+	);
+}
+
+ASTPointer<ElementaryTypeName> ASTJsonImporter::createElementaryTypeName(Json::Value const& _node)
+{
+	unsigned short firstNum;
+	unsigned short secondNum;
+	string name = member(_node, "name").asString();
+	Token token;
+	tie(token, firstNum, secondNum) = TokenTraits::fromIdentifierOrKeyword(name);
+	ElementaryTypeNameToken elem(token, firstNum,  secondNum);
+
+	std::optional<StateMutability> mutability = {};
+	if (_node.isMember("stateMutability"))
+		mutability = stateMutability(_node);
+
+	return createASTNode<ElementaryTypeName>(_node, elem, mutability);
+}
+
+ASTPointer<UserDefinedTypeName> ASTJsonImporter::createUserDefinedTypeName(Json::Value const& _node)
+{
+	vector<ASTString> namePath;
+	vector<string> strs;
+	string nameString = member(_node, "name").asString();
```

Comment:
All the `as*` helpers try to cast the json value to the requested type, but they may throw an exception.